### PR TITLE
Migrate CreateVolume response topology to standard label topology.kubernetes.io/zone

### PR DESCRIPTION
**Pull request by ConnorJC3**
*Is this a bug fix or adding new feature?**

Feature

**What is this PR about? / Why do we need it?**

See https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/729#issuecomment-1942026577

**What testing is done?** 

CI/Manual

### CODE REVIEW

**Comment by ConnorJC3** [comment]
the flake experience
/retest

File: pkg/driver/controller.go [hunk 1]
```diff
@@ -924,7 +924,7 @@ func newCreateVolumeResponse(disk *cloud.Disk, ctx map[string]string) *csi.Creat
 		}
```
Comment (AndrewSirenko) [inline pkg/driver/controller.go:924]: Question: Should we also have to change our examples documentation to the well known key like [in storage class example](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/ec8718afe87806ed5627c6fec54e54240feb5289/examples/kubernetes/storageclass/manifests/storageclass.yaml#L28)?

File: pkg/driver/controller.go [hunk 2]
```diff
@@ -924,7 +924,7 @@ func newCreateVolumeResponse(disk *cloud.Disk, ctx map[string]string) *csi.Creat
...
 	}
```
Comment (AndrewSirenko) [inline pkg/driver/controller.go:925]: Should we change our e2e tests to use new topology key, as mentioned in this TODO [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/ec8718afe87806ed5627c6fec54e54240feb5289/tests/e2e/driver/ebs_csi_driver.go#L54-L55) 
?

**Review by torredil — APPROVED** [review]
/lgtm
/approve

**Pull request merged**